lightwood/analysis/explain.py

```diff
@@ -94,18 +93,20 @@ def explain(data: pd.DataFrame,
 
             # base ICP
             X = deepcopy(icp_X)
+            # Ugly optimization but accessing `values` actually wasted a lot of time
+            icp_values = X.values
 
             # get all possible ranges
             if timeseries_settings.is_timeseries and timeseries_settings.nr_predictions > 1 and is_numerical:
 
                 # bounds in time series are only given for the first forecast
                 analysis['icp']['__default'].nc_function.model.prediction_cache = \
                     [p[0] for p in predictions['prediction']]
-                all_confs = analysis['icp']['__default'].predict(X.values)
+                all_confs = analysis['icp']['__default'].predict(icp_values)
 
             elif is_numerical:
                 analysis['icp']['__default'].nc_function.model.prediction_cache = predictions['prediction']
-                all_confs = analysis['icp']['__default'].predict(X.values)
```

Comment:
I'm assuming calling X.values twice actually increased the speed. Do you have evidence? As in, can you perform a timing test on how long it actually takes to call this function?